src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1984,10 +2125,78 @@ internal static uint GetValEscape(BoundExpression expr, uint scopeOfTheContainin
 
                     return Math.Max(GetValEscape(binary.Left, scopeOfTheContainingExpression),
                                     GetValEscape(binary.Right, scopeOfTheContainingExpression));
+
+                case BoundKind.UserDefinedConditionalLogicalOperator:
+                    var uo = (BoundUserDefinedConditionalLogicalOperator)expr;
+
+                    return Math.Max(GetValEscape(uo.Left, scopeOfTheContainingExpression),
+                                    GetValEscape(uo.Right, scopeOfTheContainingExpression));
+
+                case BoundKind.QueryClause:
```

Comment:
I didn't see tests for queries.

Reply:
I could not figure how to construct a valid case. It looks like one way or another `IEnumerable` requirement stands in the way. I kept the implementations - in case it gets hit by error case or if there is actually a way, but I think it is a low-pri to find a test for this. 

---
In reply to: [141157807](https://github.com/dotnet/roslyn/pull/22303#discussion_r141157807) [](ancestors = 141157807)

Reply:
We can track this as a separate issue. - [https://github.com/dotnet/roslyn/issues/22349](https://github.com/dotnet/roslyn/issues/22349)


---
In reply to: [141163519](https://github.com/dotnet/roslyn/pull/22303#discussion_r141163519) [](ancestors = 141163519,141157807)